.github/workflows/build-test-ship.yaml

```diff
@@ -128,18 +175,142 @@ jobs:
             ${{ env.ECR_REGISTRY }}/${{ env.ECR_REPOSITORY }}:${{ env.IMAGE_TAG }}
             ${{ env.ECR_REGISTRY }}/${{ env.ECR_REPOSITORY }}:latest
 
-  # Post release status to #deployment.
-  update-slack:
-    name: Update Slack
-    runs-on: ubuntu-latest
+  # Builds and ships Helm charts if necessary
+  push-charts:
     needs: build-and-ship-image
-    if: ${{ github.event.release }}
+    runs-on: ubuntu-latest
+    if: |
+      github.event_name == 'push' || 
+      github.event_name == 'release'
     steps:
-      - name: Post to a Slack channel
-        id: slack
+      - uses: actions/checkout@v2
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+
+      # If the chart version changed we need to reflect it in the chart(s).
+      # Also reflects app changes if both change in the same PR.
+      - name: Update Versions in Helm Chart(s)
+        id: update-versions
+        if: |
+          github.event_name == 'push' &&
+          needs.build-and-ship-image.outputs.chartchange == 'true'
+        run: |
+          . version.env
+          sed -i 's/^appVersion: .*/appVersion: \"'${APPVERSION}'\"/' charts/*/Chart.yaml
+          sed -i 's/^version: .*/version: '${CHARTVERSION}'/' charts/*/Chart.yaml
+          if ! git diff --quiet; then
+            git config --global user.name github-actions
+            git config --global user.email ""
+            git add -A
+            git commit -m "Updating Helm charts with appversion ${APPVERSION}, chartversion ${CHARTVERSION}"
+            git push
+          fi
+
+      # If this was a push to dev and we need to ship charts, ship prereleases first
+      - name: Mark as Prerelease
+        id: mark-prerelease
+        if: |
+          github.event_name == 'push' && 
+          needs.build-and-ship-image.outputs.chartchange == 'true'
+        run: |
+          sed -i '/^version/ s/$/-beta/' charts/*/Chart.yaml
+
+      # AWS Setup
+      - name: Configure AWS credentials
```

Comment:
Why are there two of these steps?

Reply:
One is in the `build-and-ship-image` job and the other is in the `push-charts` job. The jobs run in different contexts/on different runners so both should be needed.